### PR TITLE
[BACKEND-806] Modifications for internal build

### PR DESCRIPTION
**Pull request by xanec**
This modifies the build to:
1. fix the version of `joda-time` to `2.9.9` to ensure repeatable build
2. use (active) repositories configured in `settings.xml` (if available) for build

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,7 +12,32 @@ subprojects {
     group = "com.netflix.${githubProjectName}" // TEMPLATE: Set to organization of project
 
     repositories {
-        jcenter()
+        File settingsFile = new File(m2SettingsFile.replace('~', System.env.HOME.toString()))
```
Comment (drcrallen) [inline build.gradle]: should only be at start of string, yes?

File: build.gradle [hunk 2]
```diff
@@ -12,7 +12,32 @@ subprojects {
...
+        if (settingsFile.isFile()) {
+            println "Detected m2 settings file: ${settingsFile}"
```
Comment (drcrallen) [inline build.gradle]: does gradle not have a standard library that interfaces with maven?

File: gradle.properties [hunk 3]
```diff
@@ -12,3 +12,6 @@ jacksonVersion=1.8.3
 luceneVersion=3.6.0
 awsVersion=1.9.3
 mockitoVersion=1.8.5
+jodaTimeVersion=2.9.9
+
+m2SettingsFile=~/.m2/settings.xml
```
Comment (drcrallen) [inline gradle.properties]: This is not really safe. many build environments don't pull from `~/.m2` and `~/.m2` is not guaranteed to be the maven directory.

**Comment by drcrallen** [comment]
Suggest using standard libraries to interface with maven instead of writing our own.

File: gradle.properties (hunk 3 above)
Comment (xanec) [reply gradle.properties]: You can override the properties using the flag `-Pm2SettingsFile=<path-to-settings.xml>`, would this be okay?

File: build.gradle (hunk 2 above)
Comment (xanec) [reply build.gradle]: Apparently, the only way (that I know of) is to hard-code the settings (including credentials) into the build file to access custom repository.

Comment (drcrallen) [reply build.gradle]: What does `mavenLocal()` not do  that `settings.xml` does?

Comment (drcrallen) [reply build.gradle]: We really need to be using https://docs.gradle.org/current/userguide/dependency_management.html#sec:repositories if at all possible.

Comment (xanec) [reply build.gradle]: I believe `mavenLocal()` only looks to the local maven cache (i.e.., `~/.m2/cache`) but does pull form external repositories

**Comment by xanec** [comment]
Seems like there is definitely no support on Gradle-side to read configurations from `settings.xml` (refer to [this](https://issues.gradle.org/browse/GRADLE-1039) and [this](https://github.com/gradle/gradle/issues/1236)). This new commit gets pretty close by only requiring the builder to specify the `server id` in the `settings.xml` and the respective URL.

File: build.gradle [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
 subprojects {
     apply plugin: 'nebula.netflixoss'
     apply plugin: 'java'
+    apply plugin: 'net.linguica.maven-settings'
 
     group = "com.netflix.${githubProjectName}" // TEMPLATE: Set to organization of project
 
     repositories {
-        jcenter()
+        if (project.hasProperty("repoName") && project.hasProperty("repoUrl")) {
```
Comment (dgolitsyn) [inline build.gradle:21]: Where auth credentials for specified repo should be provided?

Comment (xanec) [reply build.gradle:21]: The credentials will be read from the maven's `settings.xml` file. Specifically, the value of `repoName` should points to one of the `id` tag under `settings->servers->server`, then the plugin should use the corresponding crendentials

**Pull request merged**